gcore/gdalalgorithm.cpp

```diff
@@ -267,9 +257,24 @@ bool GDALAlgorithmArg::Set(const std::string &value)
     }
 
     if (m_decl.IsRemoveSQLCommentsEnabled())
-        newValue = CPLRemoveSQLComments(newValue);
+        value = CPLRemoveSQLComments(value);
+
+    return true;
+}
+
+bool GDALAlgorithmArg::Set(const std::string &value)
+{
+    if (m_decl.GetType() != GAAT_STRING)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "Calling Set(std::string) on argument '%s' of type %s is not "
```

Comment:
prepare for localization? _()

Reply:
same as above. This should be a more general move to autoconvert all CPLError() calls if we decide to go for localization